plugins/modules/register_cluster.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, print_function
+
+import json
+import os
+import requests
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: register_cluster
+
+short_description: Register AssistedInstaller cluster
+
+version_added: "1.0.0"
+
+description: Register an AssistedInstaller Cluster
+
+options:
+    data:
+        description: JSON data for AssistedInstaller
+        type: dict
+        requried: true
```

Comment:
How would you feel by splitting the data into the minimal options, i.e.

```yaml
  name: 
    description: Name of the cluster
    type: str
    required: true
  pullsecrets: 
    description: Pullsecrets
    type: str
    required: true
  version:
    description: OpenShift version
    type: str
    required: true
```

I know this changes the interface, but has some advantages as to perform validation on the data passed.

Reply:
Hey @tonyskapunk, this was the first solution I considered. The caveat is that there are other valid fields that could be passed to data (example below), which is not nicely handled. Maybe with an "other" variable that takes in a dict, what would your suggestion be to handle this? It can be 1. A larger interface of all acceptable vars, 2. This interface and an "other" var
`"name": "testcluster",
  "openshift_version": "4.16", 1
  "high_availability_mode": "<mode>", 2
  "base_dns_domain": "example.com",
  "network_type": "examplenetwork",
  "cluster_network_cidr":"11.111.1.0/14"
  "cluster_network_host_prefix": 11,
  "service_network_cidr": "111.11.1.0/16",
  "api_vips":[{"ip": ""}],
  "ingress_vips": [{"ip": ""}],
  "vip_dhcp_allocation": false,
  "additional_ntp_source": "clock.redhat.com,clock2.redhat.com",
  "ssh_public_key": "$CLUSTER_SSHKEY",
  "pull_secret"`

Reply:
Agree, the downside is that it can grow really long the list of inputs to have. But it gives a better control on what's passed and `AnsibleModule` handles some of that already.

When @vjayaramrh and I were discussing on a similar change about the inputs, we talked about the collection kubernetes.core is doing on the creation/modification of the resources, that offers two ways for the inputs.
One is by generic fields and other is by the use of  `definition`[1] maybe it's worth that approach in here as well.

[1] https://docs.ansible.com/ansible/latest/collections/kubernetes/core/k8s_module.html#parameter-resource_definition

All, in all, I'm OK getting this one in, and improving it overtime.